src/Translator.js

```diff
@@ -13,40 +13,31 @@ var components = [
   'LED Daytime',
   'Clean Switch',
   'Bucket Switch',
-]
+];
 
 module.exports = function() {
   return {
     translateToObject: function(diagnosticsData) {
-      var context = {
-        result: {
-          diagnostics: {}
-        },
-        temporaryEntry: {}
-      };
+      var diagnostics = {}
 
-      for (var diagnosticEntryIndex = 0; diagnosticEntryIndex < Object.keys(diagnosticsData.rawDiagnostics).length; diagnosticEntryIndex++) {
-        context.entry = diagnosticsData.rawDiagnostics[diagnosticEntryIndex];
-        components.forEach(component, context);
-        context.result.diagnostics[diagnosticEntryIndex] = context.temporaryEntry;
```

Comment:
Note: assigning one object to another in JS just points to the other object, it doesn't copy the contents.